src/plugins/data_source_management/public/components/data_source_view/data_source_view.tsx

```diff
@@ -111,11 +116,22 @@ export class DataSourceView extends React.Component<DataSourceViewProps, DataSou
           this.props.onSelectedDataSources([{ id: optionId, label: selectedDataSource.title }]);
         }
       } catch (error) {
-        handleDataSourceFetchError(
-          this.onError.bind(this),
-          this.props.notifications!,
-          this.props.onSelectedDataSources
-        );
+        // return [] to customer when no default data source
+        if (!defaultDataSource) {
+          this.props.onSelectedDataSources?.([]);
```

Comment:
when there is error to get the data source, do we return empty array and when do we return something else?

Reply:
https://github.com/opensearch-project/OpenSearch-Dashboards/pull/6610#discussion_r1576973918, should also check if the local cluster is hide or not